astropy/cosmology/parameter.py

```diff
@@ -0,0 +1,265 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import astropy.units as u
+
+__all__ = ["Parameter"]
+
+
+class Parameter:
+    r"""Cosmological parameter (descriptor).
+
+    Should only be used with a :class:`~astropy.cosmology.Cosmology` subclass.
+
+    Parameters
+    ----------
+    fget : callable[[type], Any] or None, optional
+        Function to get the value from instances of the cosmology class.
+        If None (default) returns the corresponding private attribute.
+        Often not set here, but as a decorator with ``getter``.
+    fvalidate : callable[[object, object, Any], Any] or {'default', 'float'}, optional
```

Comment:
List of pre-defined validators is incomplete... My tendency would be to just write `str` here.

Reply:
SGTM. I'm adding a property ``registered_validators(self): return self._registry_validators.keys()`` so it's easy to discover what's a valid str.

Reply:
made it a classproperty

Reply:
Done!